### PR TITLE
Some Balance

### PR DESCRIPTION
**Pull request by DarkoniusXNG**
TLDR: Nerfed top picks at the Minor. Buffed worst heroes.
Most picked/banned heroes at the Minor: Ogre Magi, Grimstroke, Centaur, Undying, Witch Doctor, Zeus, Riki, Bloodseeker, Viper, Medusa, AA, Kunkka, Weaver, Pangolier, Jakiro, Axe, Dazzle, Elder Titan, Spirit Breaker, Ursa, Sniper, Spectre, Io, Troll, Wraith King and Lich.

This PR includes:
1) Antimage's mana break damage nerf. He needed some dmg nerf in the lategame.
2) Ogre Magi nerfs because he was additionally buffed in 7.20.
3) Witch Doctor rebalance - stronger early, nerfed talent that gives huge power spike.
4) Riki rebalance - nerfing annoying parts of Riki (Smoke Screen cd), Ult rebalance. Sohei now stronger than Riki.
5) Viper reasonable nerfs because she is a top pick for months.
6) Elder Titan nerfs because he was additionally buffed in 7.20.
7) Shadow Demon fixing the overpowered spell cd.
8) Chaos Knight small nerfs here and there because he is annoying to play against.
9) AA mana regen talent buff.
10) Chen buffs.
11) Phantom Assassin Blur active effect rebalance so its less useful for farming or killing bosses.
12) Slardar buffs because he is one of the worst heroes and we didn't add the new Bash of the Deep.
13) Underlord small buff.
14) Butterfly nerf and Diffusal Blade buff because every agility hero buys Butterfly.

Other thoughts:
1) Grimstroke doesn't need a nerf. He was nerfed by Valve multiple times. He is still strong but not overpowered.
2) Centaur: Centaur's return was changed in 7.20. So he was nerfed indirectly.
3) Undying needs a nerf (probably bonus hp during his ult).
4) Zeus maybe needs a small mana cost nerf.
5) Dazzle doesn't need a nerf or a buff. He is one of the best supports right now.
6) Ursa was nerfed several times by Valve but he is still strong because of the nature of the game.
7) Sniper's knockback talent is the most annoying thing in the game.
8) Spectre was heavily nerfed by Valve (Dispersion, Desolate). She doesn't need more.
9) Troll Warlord is in a weird place right now. Strong because of the nature of the game but can easily be kited, disarmed etc.
10) Wraith King Aghanim Scepter effect second most annoying thing in the game. If we ever decide to nerf him, best nerf would be dmg reduction while in wraith form.

### CODE REVIEW

File: game/scripts/npc/items/item_butterfly.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@
       "01"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "bonus_agility"                                   "35 45 55 65 75"
+        "bonus_agility"                                   "35 40 45 50 75"
```
Comment (chrisinajar) [inline game/scripts/npc/items/item_butterfly.txt]: why did you make butterfly scale really weirdly?

File: game/scripts/npc/items/item_diffusal_blade.txt [hunk 2]
```diff
@@ -67,7 +67,7 @@
       "01"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "bonus_agility"                                   "20 30 45 70 105"
+        "bonus_agility"                                   "20 40 45 70 105"
```
Comment (chrisinajar) [inline game/scripts/npc/items/item_diffusal_blade.txt]: same with this, it's really weird scaling..
